File: python/src/st_course.py
#
# STAR TREK Text Game
# 2017-05-01 K.OHWADA
#
	
## class
class Course():
	
	COURSE_DELTA = [ [0,1], [-1,1], [-1,0], [-1,-1], [0,-1], [1, -1], [1,0], [1,1] ]
	
	COURSE_RIGHT = 1
	COURSE_UP_RIGHT = 2
	COURSE_UP = 3
	COURSE_UP_LEFT = 4
	COURSE_LEFT = 5
	COURSE_DOWN_LEFT = 6
	COURSE_DOWN = 7
	COURSE_DOWN_RIGHT = 6
																		
										
	def __init__(self):
		pass				
### def end
	
	def get_course_delta(self, course):
		delta = [0,0]
		if ( course >= 1 ) and ( course <= 8 ):
			delta = self.COURSE_DELTA[course-1]
		return delta
### def end
	
	def get_course(self, x_orig, y_orig, x, y):
		ret = 0
		abs_x = abs( x_orig - x )
		abs_y = abs( y_orig - y )
		if ( x_orig == x ) and ( y_orig < y ):
			ret = self.COURSE_RIGHT 
		elif ( x_orig == x ) and ( y_orig > y ):
			ret = self.COURSE_LEFT
		elif ( y_orig == y ) and ( x_orig < x ):				
			ret = self.COURSE_DOWN
		elif ( y_orig == y ) and ( x_orig > x ):				
			ret = self.COURSE_UP	
		elif ( abs_x == abs_y ) and ( x_orig > x ) ( y_orig > y ) :		
			ret = self.COURSE_UP_LEFT
		elif ( abs_x == abs_y ) and ( x_orig > x ) ( y_orig < y ):	
			ret = self.COURSE_UP_RIGHT
		elif ( abs_x == abs_y ) and ( x_orig < x ) ( y_orig > y ) :		
			ret = self.COURSE_DOWN_LEFT						
		elif ( abs_x == abs_y ) and ( x_orig < x ) ( y_orig < y ) :		
			ret = SELF.COURSE_DOWN_RIGHT	 
		return ret
### def end	
								
### class end

File: javascript/src/js/st_course.js
//
// STAR TREK Text Game
// 2017-06-01 K.OHWADA
//

var Course = function () {	

	
	this.COURSE_DELTA = [ [0,1], [-1,1], [-1,0], [-1,-1], [0,-1], [1, -1], [1,0], [1,1] ];
	
	this.COURSE_RIGHT = 1;
	this.COURSE_UP_RIGHT = 2;
	this.COURSE_UP = 3;
	this.COURSE_UP_LEFT = 4;
	this.COURSE_LEFT = 5;
	this.COURSE_DOWN_LEFT = 6;
	this.COURSE_DOWN = 7;
	this.COUR_DOWN_RIGHT = 8;
	
	} // constructor										
	
	Course.prototype.get_delta = function (course) {
		var delta = [0,0]
		if (( course >= 1 ) && ( course <= 8 )) {
			delta = this.COURSE_DELTA[course-1];
		}
		return delta
		
	} // get_delta
		

Course.prototype.get_course = function (x_orig, y_orig, x, y) {	

		var ret = 0;
		var abs_x = Math.abs( x_orig - x );
		var abs_y = Math.abs( y_orig - y );
		
		if (( x_orig == x ) && ( y_orig < y )) {
			ret = this.COURSE_RIGHT;
			 
		} else if (( x_orig == x ) && ( y_orig > y )) {
			ret = thiis.COURSE_LEFT;
			
		} else if (( y_orig == y ) && ( x_orig < x )) {				
			ret = this.COURSE_DOWN;
			
		} else if (( y_orig == y ) && ( x_orig > x )) {				
			ret = this.COURSE_UP	
		} else if (( abs_x == abs_y ) && ( x_orig > x ) && ( y_orig > y )) {		
			ret = this.COURSE_UP_LEFT;
			
		} else if (( abs_x == abs_y ) && ( x_orig > x ) &&  ( y_orig < y )) {	
			ret = this.COURSE_UP_RIGHT;
			
		} else if (( abs_x == abs_y ) && ( x_orig < x ) && ( y_orig > y )) {		
			ret = self.COURSE_DOWN_LEFT	;
								
		} else if (( abs_x == abs_y ) && ( x_orig < x ) && ( y_orig < y )) {		
			ret = this.COURSE_DOWN_RIGHT;
			
	} // if
		 
		return ret;

} //get_course
